realm/realm-library/src/main/java/io/realm/RealmInstanceCallback.java

```
@@ -0,0 +1,98 @@
+ *
+ *   private Realm realm = null;
+ *   private RealmAsyncTask realmAsyncTask;
+ *
+ *   \@Override
+ *   protected void onCreate(Bundle savedInstanceState) {
+ *     super.onCreate(savedInstanceState);
+ *     setContentView(R.layout.layout_main);
+ *     realmAsyncTask = Realm.getDefaultInstanceAsync(new RealmInstanceCallback<Realm>() {
+ *         \@Override
+ *         public void onSuccess(Realm realm) {
+ *             if (isDestroyed()) {
+ *                 // If the activity is destroyed, the Realm instance should be closed immediately to avoid leaks.
+ *                 realm.close();
+ *             } else {
+ *                 MainActivity.this.realm = realm;
+ *                 // Remove the spinner and start the real UI.
+ *             }
+ *         }
+ *     });
+ *
+ *     // Show a spinner before Realm instance returned by the callback.
+ *   }
+ *
+ *   \@Override
+ *   protected void onDestroy() {
+ *     super.onDestroy();
+ *     if (realm != null) {
+ *         realm.close();
+ *         realm = null;
+ *     } else if (realmAsyncTask != null) {
+ *         // Calling cancel() will not guarantee that the task will stop delivery.
+ *         // Needs to close the Realm instance in the onSuccess() callback if it doesn't.
+ *         realmAsyncTask.cancel();
+ *     }
+ *   }
+ * }
+ * </pre>
+ *
+ * @param <T> {@link Realm} or {@link DynamicRealm}.
+ */
+public abstract class RealmInstanceCallback<T extends BaseRealm> {
+
+    /**
+     * Deliver a Realm instance to the caller thread.
+     *
+     * @param realm the Realm instance for the caller thread.
+     */
+    public abstract void onSuccess(T realm);
+
+    /**
+     * Deliver the error happens when creating the Realm instance to the caller thread. The default implementation will
+     * throw the exception on the caller thread.
+     *
+     * @param exception happens when creating the Realm instance.
+     */
+    public void onError(RuntimeException exception) {
```

Background threads should never swallow exceptions IMO. This should be `Throwable`